core/src/main/scala/org/apache/spark/serializer/SerializerManager.scala

```diff
@@ -103,16 +108,44 @@ private[spark] class SerializerManager(defaultSerializer: Serializer, conf: Spar
   }
 
   /**
+   * Wrap an input stream for encryption and compression
+   */
+  def wrapStream(blockId: BlockId, s: InputStream): InputStream = {
+    wrapForCompression(blockId, wrapForEncryption(s))
+  }
+
+  /**
+   * Wrap an output stream for encryption and compression
+   */
+  def wrapStream(blockId: BlockId, s: OutputStream): OutputStream = {
+    wrapForEncryption(wrapForCompression(blockId, s))
```

Comment:
why not compress first then encrypt? The performance will be better since it encrypts less data.